pkg/tasks/nodeexporter.go

```diff
@@ -117,6 +118,26 @@ func (t *NodeExporterTask) Run(ctx context.Context) error {
 		return errors.Wrap(err, "initializing node-exporter ServiceMonitor failed")
 	}
 
+	err = t.client.CreateOrUpdateServiceMonitor(ctx, smn)
+	if err != nil {
+		return errors.Wrap(err, "reconciling node-exporter ServiceMonitor failed")
+	}
+
+	smn, err = t.factory.NodeExporterOperationalServiceMonitor()
```

Comment:
maybe have a `t.factory.NodeExporterServiceMonitors()` method which would return a `map[string]*monv1.ServiceMonitor` to reduce the boiler-plate code? 
The same comment applies to the other tasks.